src/clsourcedependencies_parser.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_CLSOURCEDEPENDENCIESPARSER_H_
+#define NINJA_CLSOURCEDEPENDENCIESPARSER_H_
+
+#include <set>
+#include <string>
+
+/// This class parses the output of of the /sourceDependencies flag from Visual
+/// Studio's cl.exe, which generates a JSON file with dependency information.
+struct CLSourceDependenciesParser {
+  /// Parse the JSON file produced by /sourceDependencies, filling includes_
+  /// with the includes. Returns true on success, or false with err filled.
+  bool Parse(const std::string& content, std::string* err);
+
+  /// Set of full paths to the parsed includes.
+  std::set<std::string> includes_;
```

Comment:
This seems weird.

I recommend instead either

1. returning the result from the Parse function.
2. Allowing for the storage for the result to be passed into the Parse function as, either a reference to the set, or a std::back_inserter or something

Reply:
Changed from a class to a `ParseClSourceDependencies()` free function that accepts a `set<string>&` for outputting include paths.